plugin/core/url.py

```diff
@@ -47,6 +48,14 @@ def uri_to_filename(uri: str) -> str:
         return url2pathname(parsed.path)
 
 
+def path_url_to_fs_path(path: str) -> str:
```

Comment:
How about instead having a function that returns a tuple (scheme, filepath) where filepath is None for non-file schemas?

There is a lot of repetition right now, having to parse manually using some low-level functions.

Reply:
I think this is a good idea, went with 3194319